### PR TITLE
Add geant setup options to configure bins per decade

### PR DESCRIPTION
**Pull request by sethrj**

### CODE REVIEW

File: src/io/GeantSetupOptionsIO.json.cc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+namespace celeritas
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+// HELPER FUNCTIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Get a string corresponding to a surface type.
```
Comment (stognini) [inline src/io/GeantSetupOptionsIO.json.cc:23]: "... corresponding to the physics list value."

File: src/io/GeantSetupOptionsIO.json.cc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+{
+    opts.physics = gspl_from_string(j.at("physics").get<std::string>());
+    j.at("em_bins_per_decade").get_to(opts.em_bins_per_decade);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Write options to JSON.
+ */
+void to_json(nlohmann::json& j, const GeantSetupOptions& opts)
```
Comment (stognini) [inline src/io/GeantSetupOptionsIO.json.cc:82]: I have a branch that is storing EM parameters in our ImportData as a part of the of the solution for having equivalent physics in both G4 and Celeritas. So when that's merged, we should be able to retrieve the number of bins from the ROOT file too.

File: src/io/GeantSetupOptionsIO.json.cc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    static_assert(
+        static_cast<int>(GeantSetupPhysicsList::size_) * sizeof(const char*)
+            == sizeof(strings),
+        "Enum strings are incorrect");
+
+    return strings[static_cast<int>(value)];
+}
+
+//---------------------------------------------------------------------------//
+// TODO: template and rewrite these as a helper class
```
Comment (stognini) [inline src/io/GeantSetupOptionsIO.json.cc:43]: I assume this is going to be tackled at another time.

**Comment by stognini** [comment]
Can't leave a comment on that line, so here it is. Seems like jenkins is unhappy because it seems that 

`GeantSetup.hh`
```c++
//---------------------------------------------------------------------------//
#if !CELERITAS_USE_GEANT4
inline GeantSetup::GeantSetup(const std::string&, PhysicsList)
```

should probably be

```c++
//---------------------------------------------------------------------------//
#if !CELERITAS_USE_GEANT4
inline GeantSetup::GeantSetup(const std::string&, Options)
```

**Comment by stognini** [comment]
Aside from the few minor comments, it looks great.

**Review by stognini — APPROVED** [review]

File: src/io/GeantSetupOptionsIO.json.cc (hunk 3 above)
Comment (sethrj) [reply src/io/GeantSetupOptionsIO.json.cc:43]: Yep! It works for the current need but when we want to add enum class I/O somewhere else, this is the "TODO" for where it'll be generalized :)

**Pull request merged**